src/main/java/io/github/moulberry/notenoughupdates/util/Utils.java

```diff
@@ -254,46 +254,40 @@ public static String shortNumberFormat(double n, int iteration) {
                 : shortNumberFormat(d, iteration + 1));
     }
 
-    public static String trimIgnoreColour(String str) {
-        return trimIgnoreColourStart(trimIgnoreColourEnd(str));
-    }
-
-    public static String trimIgnoreColourStart(String str) {
-        str = str.trim();
-        boolean colourCodeLast = false;
-        StringBuilder colours = new StringBuilder();
-        for (int i = 0; i < str.length(); i++) {
-            char c = str.charAt(i);
-            if (colourCodeLast) {
-                colours.append('\u00a7').append(c);
-                colourCodeLast = false;
+    public static String trimWhitespaceAndFormatCodes(String str) {
```

Comment:
I am not sure about intended behaviour, but to me it looks like the old `trimIgnoreColour` trims whitespace while keeping color codes in front of said white space, while the new behaviour removes those color codes. I also think some of the other code wanted that old behaviour (like, the auction house which uses this function to keep rarity colours for pets)

Reply:
Doesnt mean the new function cant be useful *as well*, maybe both can remain.

Reply:
I agree that both methods are useful. I have restored the old methods and switched appropriate usages back to trimIgnoreColour.

FWIW - by intended behavior, I was referring to the original behavior expected by the mining code that was present prior to [3290259e](https://github.com/Moulberry/NotEnoughUpdates/commit/3290259e39e9816fd92afa85710de01a6ddfa3b9).